internal/controllers/configuration/secret_controller.go

```diff
@@ -0,0 +1,119 @@
+package configuration
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/dataplane"
+	"github.com/kong/kubernetes-ingress-controller/v2/internal/util"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+// -----------------------------------------------------------------------------
+// CoreV1 Secret - Reconciler
+// -----------------------------------------------------------------------------
+
+// CoreV1SecretReconciler reconciles Secret resources
+type CoreV1SecretReconciler struct {
+	client.Client
+
+	Log             logr.Logger
+	Scheme          *runtime.Scheme
+	DataplaneClient *dataplane.KongClient
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *CoreV1SecretReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	c, err := controller.New("CoreV1Secret", mgr, controller.Options{
+		Reconciler: r,
+		LogConstructor: func(_ *reconcile.Request) logr.Logger {
+			return r.Log
+		},
```

Comment:
Please make sure to pass down the `CacheSyncTimeout` as introduced in https://github.com/Kong/kubernetes-ingress-controller/pull/3013.